frontend/device/cervantes/device.lua

```diff
@@ -21,6 +21,14 @@ local function isConnected()
     return carrier
 end
 
+local function isMassStorageSupported()
+    -- we rely on 3rd party package for that. It should be installed as part of KOReader prerequisites,
+    local safemode_version = io.open("/usr/share/safemode/version", "rb")
+    if not safemode_version then return false end
```

Comment:
Btw, it doesn't make much of a difference but if you're not actually reading the file we also have this utility function:

https://github.com/koreader/koreader/blob/35e5c24cb8772e3f78c2909544192c144219de61/frontend/util.lua#L396-L402

Reply:
As you like, I can also use lfs.attributes directly. Whatever fits better :smile:

Reply:
The advantage of using an abstraction is twofold:

1. You can easily change the underlying implementation. That is, I wouldn't use LFS directly for this purpose. (Of course you might find that older code does it at least a few times, including some written by myself iirc.)
2. You can potentially get a clearer overview of where it's actually used.

That being said, the main reason for the existence of these utility functions is that LFS errors instead of returning something like false or nil so let's just leave it as it is.